app/pgbouncer/pgbouncer_manager.py

```diff
@@ -10,8 +9,44 @@ def __init__(self):
         super().__init__()
         self.pgbouncer_process = None
         self.neon_api = NeonAPI()
+        self.cert_path = "/etc/pgbouncer/server.crt"
+        self.key_path = "/etc/pgbouncer/server.key"
+
+    def _generate_certificates(self):
+        """Generate self-signed certificates if they don't exist."""
+        if os.path.exists(self.cert_path) and os.path.exists(self.key_path):
+            return
+
+        print("Generating self-signed certificates...")
+        # Generate private key
+        subprocess.run([
+            "openssl", "genrsa", "-out", self.key_path, "2048"
+        ], check=True)
+        
+        # Generate CSR
+        subprocess.run([
+            "openssl", "req", "-new", "-key", self.key_path,
+            "-out", "/tmp/server.csr",
+            "-subj", "/CN=localhost"
```

Comment:
Would be great to add `O=DO NOT TRUST/OU=Neon Local self-signed cert`, to clearly signal this is a temporary cert and not to be trusted.

https://www.crunchydata.com/blog/improving-pgbouncer-security-with-tlsssl